### DISCUSSION TITLE
Regrading Map Distribution

### DISCUSSION THREAD

**Discussion by ajithcodesit** [VDA5050/VDA5050#302] [Q&A]
Hello,

I was going through the new VDA5050 v2.1.0 release candidate and saw the map distribution part. A map server is introduced to store a set of maps and with instant actions the maps can be downloaded to and enabled on the AGV. I have the following doubts regarding the this:

1. The format of the map data. Is it a PNG, PGM or JPEG image?

2. Assuming that the map is an image, some additional metadata like map origin with respect to the image and resolution of the map in meters per pixel might be required. What should be the format for this?

3. Should there be an instant action to upload a map from the vehicle to the map server? If the vehicle is an AMR with mapping capability using SLAM, the generated map could be uploaded to the map server under master control's discretion.

4. When considering AMR with self localisation capability, different vehicles might require different maps to effectively keep it's localisation. This might due to the placement of the LiDAR with respect to the vehicle's body.

5. The map server download procedure is not clear. Is it a normal FTP server?

Thanks in advance.

**Comment by jabbenseth** [comment]
Hi, 
this is not an official answer but following the discussions and of the top my mind. Also while our implementation works with our server and vehicle and was implemented along the discussions, it was never tested with third party vehicles or third party master control systems. I happily learn where our assumptions are wrong in terms of sticking to VDA5050 or about other interpretations and implementations.

1. The format of maps was intentionally not standardized due to the huge number of variations for different localisation and planning technologies as well as different implementations of these.
2. Given 1. : that's your choice.
3. As far as implementation of vda5050 goes, the set of instant actions is quite often extended by custom actions that aren't mentioned in the document. Uploading a map can be one such action.
4. Yes, given my answer to 1. Your implementation can decide
5. The protocol should be part of the URI

Concluding: the standard only defines how to download but not what.

A few words on our implementation: 

Our exported maps are one Json document for metadata as well as a set of "floorplans" (arbitrary format but usually vector images, raster images or proprietary binaries) that are referenced by the Json document. All of this is hosted on a Webserver that offers RESTful access to the resources and an export to compressed tarballs. We picked HTTP because the maps are used for the Frontend (webapp) as well.

The vehicles know which floorplans to pick (according to height of the scanners and algorithms used) 
A map download is either lazy (metadata and the necessary floorplans) or complete (the exported tarball) depending on configuration, so one specific URI is sufficient but there might be multiple requests. 

For map upload we have two cases: if the user is "mapping" (initial or extending/augmenting an existing floorplan) we don't use VDA5050 but our Frontend guides the process, postprocessing and finally uploading. For collaborative liveslam we have implemented an instant action (close to the map download definition) that is used to upload differential maps into the fusion service.

**Reply by ajithcodesit** [reply]
Thanks for your detailed reply. 

Another doubt that came up while reading your implementation details is that, if you have multiple maps, there is no common coordinate that relates these maps together. 

For example, robots with scanner at different height might see different parts of the world but for planning on the master control only one map exist. So is there a way to register the different map together in you application so that they can be bought to a single coordinate? Is this what being done on the frontend as a postprocessing step?
